### PR TITLE
feat: configurable timeouts

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/oracle-node/src/utils/blockchain-providers.ts [hunk 1]
```diff
@@ -12,12 +12,16 @@ const produceProvider = (
     const rpcProviders = rpcUrls.map(
       (rpcUrl) => new providers.StaticJsonRpcProvider(rpcUrl, network)
     );
-    return new ProviderWithAgreement(rpcProviders);
+    return new ProviderWithAgreement(rpcProviders, {
+      singleProviderOperationTimeout: config.agreementProviderSingleRpcTimeout,
+      allProvidersOperationTimeout: config.agreementProvidersTimeout
```
Comment (Lukasz2891) [inline packages/oracle-node/src/utils/blockchain-providers.ts:17]: Idk if it's a good pattern, but maybe just to pass the 'config()' here? It implements all of the required fields.

**Review by rpiszczatowski — DISMISSED** [review]
LGTM

File: packages/oracle-node/src/utils/blockchain-providers.ts (hunk 1 above)
Comment (koonopek) [reply packages/oracle-node/src/utils/blockchain-providers.ts]: I don't see advantages, can lead to bug

**Review by rpiszczatowski — APPROVED** [review]

File: packages/oracle-node/src/config.ts [hunk 2]
```diff
@@ -259,4 +261,16 @@ export const config: NodeConfig = Object.freeze({
...
+      "AGREEMENT_PROVIDER_TIMEOUT",
+      DEFAULT_AGREEMENT_PROVIDERS_TIMEOUT
+    )
+  ),
+  agreementProviderSingleRpcTimeout: parserFromString.number(
+    getFromEnv(
+      "AGREEMENT_PROVIDER_SINGLE_RPC_TIMEOUT",
+      DEFAULT_AGREEMENT_PROVIDER_SINGLE_RPC_TIMEOUT
+    )
+  ),
```
Comment (cehali) [inline packages/oracle-node/src/config.ts:275]: Why names are different than in relayer and agreement provider params `singleProviderOperationTimeout`, `allProvidersOperationTimeout`?

**Pull request merged**